### PR TITLE
Include range of attributes in parent (implementation files)

### LINKED ISSUE #11486: Attributes not included in the range of the parent node
Following up on previous AST related issues.
I still found some constructs where the range of the attributes is not included in the parent node.

```fsharp
[<Foo>] // SynModuleDecl.Let / SynBinding
let a = 0

[<Foo2>] // SynTypeDefn
type Bar =
    [<Foo4>] // SynMemberDefn.LetBindings / SynBinding
    let x = 8
    [<Foo2>]  // SynMemberDefn.Member / SynBinding
    member this.Something () = ()

[<Foo5>] // SynModuleDecl.NestedModule
module Nested =
    ()
```

See [AST Viewer](https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWQDaAPMa6wB8AXTAB6MWADKAT2QBZdgFcasACKxmNAHQAZWABdx0uQCEU0FAHMAOgCcbyVUapgAvGAAM9xz%2BS9%2BVgAmEXFJWWQAFRkAB3UEZD8DWNgwUyo7dz8ICACBABZQiRMFWABbACNYOw14ZD1Dc2RLZCs0YwimltsHZBywZzAAD3cwAA5snLzg0LCS%2BXKqmoTtBcrqjrMLa0mIMsWNgwALIm0pVn3j6zAACgBKUfu-P2mAViLwuUVoFXitbQAcugDLBoN9fn4yspVGAgWgQdAsr1%2Bk9epAQABfIA).

**Expected behaviour**

Attributes are included.

**Actual behaviour**

They are not.

**Known workarounds**

Parse the tree to construct a correct range.

**Related information**

FCS 39.

### PR DESCRIPTION
**Pull request by nojaf**
Fixes #11486.
In this PR I tried to take attributes into account when constructing the range of SynBinding and others.
Most likely some tests outside of FCS might break due to corrected ranges.
I'll tackle those if they pop up.

### CODE REVIEW

File: tests/service/StructureTests.fs [hunk 1]
```diff
@@ -651,11 +651,11 @@ type T() =
 """
     => [ (2, 5, 11, 12), (2, 7, 11, 12)
          (3, 4, 4, 12), (3, 7, 4, 12)
-         (3, 4, 4, 12), (4, 8, 4, 12)
+         (3, 4, 4, 12), (3, 10, 4, 12)
```
Comment (nojaf) [inline tests/service/StructureTests.fs:654]: @auduchinok I winged this test a bit 😅.
In the parser I updated the range of `SynMemberDefn.Member(SynBinding (None, SynBindingKind.Normal, false, false, $1, grabXmlDoc(parseState, 3), valSynData, declPat, None, expr, mWholeBindLhs, DebugPointAtBinding.NoneAtInvisible), m)` (see line `1988`) to only included the `lhs` of the SynBinding and not the entire range `m`.
This is the case with all the other SynBindings, so I thought I could fix this and make it consistent.

So, that leads to a change in `getOutliningRanges` (in `ServiceStructure.fs`) where at this point I'm not really sure what I did is correct. Any thoughts here?

Comment (auduchinok) [reply tests/service/StructureTests.fs:654]: We ignore some of these ranges in ServiceStructure in Rider, and from a glance it looks like it won't affect us. Not sure about other users, though.

**Comment by nojaf** [comment]
Ready for review.

**Review by cartermp — APPROVED** [review]
Thanks!

**Comment by cartermp** [comment]
I believe this should also fix a problem in the quick fixer that removes unused bindings where the attribute(s) could be left dangling.

**Pull request merged**